### PR TITLE
support trimming prefix on prometheus handler

### PR DESCRIPTION
**Pull request by achille-roussel**
The intent of this pull request is to provide a more idiomatic integration with prometheus when using defaults, by ignoring the root engine's namespace.

I didn't want to just ignore the metric namespace in the prometheus handler because I'm sure we have code that uses sub-engines to compose the full metric name
(e.g. `<service>.<category>.<metric>`).

Please take a look and let me know if anything should be changed.

### CODE REVIEW

File: prometheus/handler.go [hunk 1]
```diff
@@ -71,6 +85,13 @@ func (h *Handler) timeout() time.Duration {
 
 // ServeHTTP satsifies the http.Handler interface.
 func (h *Handler) ServeHTTP(res http.ResponseWriter, req *http.Request) {
+	switch req.Method {
```
Comment (f2prateek) [inline prometheus/handler.go]: This is unrelated to trimming right?

Comment (achille-roussel) [inline prometheus/handler.go]: Yep unrelated to trimming, but related to having defaults that do the right thing.

**Comment by achille-roussel** [comment]
Thanks for the review!
